Objects/longobject.c

```diff
@@ -4565,11 +4593,22 @@ _PyLong_Rshift(PyObject *a, size_t shiftby)
 static PyObject *
 long_lshift1(PyLongObject *a, Py_ssize_t wordshift, digit remshift)
 {
-    /* This version due to Tim Peters */
     PyLongObject *z = NULL;
     Py_ssize_t oldsize, newsize, i, j;
     twodigits accum;
 
+    if (wordshift == 0 && IS_MEDIUM_VALUE(a) &&
+        (a->ob_digit[0] & ~(PyLong_MASK >> remshift)) == 0) {
+        stwodigits sval = medium_value(a) << remshift;
```

Comment:
Isn't this is undefined behavior as well?

Reply:
[PyLong_MASK](https://github.com/python/cpython/blob/f4c03484da59049eb62a9bf7777b963e2267d187/Include/cpython/longintrepr.h#L64) is an unsigned value (defined with type `digit`) as the mask of `PyLong->ob_digit`. Maybe it can be regarded as a special `ob_digit` value somewhat.

Reply:
I mean if medium_value(a) is negative

Reply:
Thanks. Fixed.